src/components/modals/NewNodeModal/NewNodeModal.tsx

```diff
@@ -0,0 +1,168 @@
+/* eslint-disable jsx-a11y/no-static-element-interactions */
+/* eslint-disable jsx-a11y/click-events-have-key-events */
+import { ChangeEvent, useCallback, useContext, useState } from 'react'
+
+import { ModalWrapper } from 'src/components/wrappers/ModalWrapper'
+import { TextField, Button, BarGraphNode, PieGraphNode } from 'src/components'
+import { close, bar, pie, scatter, line } from './icons'
+
+import styles from './NewNodeModal.module.scss'
+import { defaultNodeData } from 'src/constants'
+import { useCreateNodeMutation } from 'src/graphql/hooks'
+import { ApolloError } from '@apollo/client'
+import { ModalContext } from 'src/contexts'
+import { useRouter } from 'next/router'
+
+const initialData = {
+  graphName: '',
+  description: '',
+  graphType: 'BAR',
+}
+
+export const NewNodeModal = () => {
+  const [data, setData] = useState({ ...initialData })
+  const {
+    useNewNodeModalState: [, setNewNodeModalState],
+  } = useContext(ModalContext)
+  const [createNode] = useCreateNodeMutation()
+  const { push } = useRouter()
+
+  const handleNameChange = useCallback(
+    (e: ChangeEvent<HTMLInputElement>) => {
+      setData({ ...data, graphName: e.target.value })
+    },
+    [data]
+  )
+
+  const handleDescriptionChange = useCallback(
+    (e: ChangeEvent<HTMLTextAreaElement>) => {
+      setData({ ...data, description: e.target.value })
+    },
+    [data]
+  )
+
+  const handleCreateNode = useCallback(async () => {
+    const title = data.graphName
+    const description = data.description
+    try {
+      await createNode({
+        variables: {
+          createNodeInput: {
+            data: JSON.stringify({
+              ...defaultNodeData,
+              title: title,
+              description: description,
```

Comment:
the createNodeInput should be of form: 
```ts
{
  title: string
  description: string
  data: string
}
```

u passed in 

```
{
  data: string
}
```

Reply:
I think title and description are in the data part, just updated the useCreateNodeMutation